api/src/test/java/org/openmrs/module/fhir2/api/search/AllergyIntoleranceSearchQueryImplTest.java

```diff
@@ -760,4 +764,91 @@ public void searchForAllergies_shouldHandleComplexQuery() {
 		    equalTo(PATIENT_UUID));
 	}
 	
+	@Test
+	public void shouldReturnCollectionOfAllergiesSortedBySeveritySevere() {
+		
+		SortSpec sort = new SortSpec();
+		sort.setParamName(AllergyIntolerance.SP_SEVERITY);
+		sort.setOrder(SortOrderEnum.ASC);
+		
+		IBundleProvider results = search(
+		 new SearchParameterMap().addParameter(FhirConstants.SEVERITY_SEARCH_HANDLER, FhirConstants.GLOBAL_PROPERTY_SEVERE, new TokenAndListParam().addAnd(new TokenParam(SEVERITY_SEVERE))).setSortSpec(sort));
+		
+		List<AllergyIntolerance> fullResults = get(results).stream().map(p -> (AllergyIntolerance)p).collect(Collectors.toList());
+		for (int i = 1; i < fullResults.size(); i++) {
+			
+			assertThat(((AllergyIntolerance) fullResults.get(i - 1)).getReactionFirstRep().getSeverity(),
+				    lessThanOrEqualTo(((AllergyIntolerance) fullResults.get(i)).getReactionFirstRep().getSeverity()));	
```

Comment:
Do you think we need to cast it again to AllergyIntolerance given that the list contains AllergyIntolerance objects?
Please use this for all other sorting tests added.